linode/manager
# refactor: [M3-8771] - Add more customization to legends and charts

## Description 📝
Problem: The height of the Recharts graph changes when the number of legend rows varies, which affects the graph size, especially with many legend rows.

Request: We would like flexibility to control the styling of legend rows and the graph itself so that it can handle dynamically varying legend rows effectively.

## Changes  🔄
- Legends should have an overflow scroll
- Legend TableHead should remain fixed when scrolling
- `MetricsDisplay`: Added `hasFixedLegendHeight` so legends across metrics should have option to have fixed height (so charts remain similar size)
- Remove double borders from legend table
- Allow ability to change `AreaChart` height and width while maintaining sensible defaults (100%)
- Fixed all the negative margin styling we were doing
- Legends should be able to account for long titles

## Target release date 🗓️
N/A

## Preview 📷
| Before  | After   | After (Fixed)
| ------- | ------- | ------- |
| <img width="1261" alt="Screenshot 2024-10-22 at 11 23 28 PM" src="https://github.com/user-attachments/assets/aed02bfd-2cb4-4615-84b9-3d61f010798c"> Current version never fills entire space until legend grows which seemed weird. | <img width="1266" alt="Screenshot 2024-10-22 at 11 23 03 PM" src="https://github.com/user-attachments/assets/7256629b-25fb-494f-a53b-5b3804429089"> | <img width="1264" alt="Screenshot 2024-10-22 at 11 24 40 PM" src="https://github.com/user-attachments/assets/f0746d50-b789-4f96-9ced-caeccec3fa4d"> |

| Before  | After   | 
| ------- | ------- | 
| ![Screenshot 2024-10-22 at 11 41 08 PM](https://github.com/user-attachments/assets/647edf2f-214a-46de-b0d7-7cf4aeb505ca) | ![Screenshot 2024-10-22 at 11 41 41 PM](https://github.com/user-attachments/assets/618565aa-4c90-462e-8258-bd89cc6640c6) |

## How to test 🧪

### Verification steps
- Go to Linode details `analytics` tab and verify charts still good (/linodes/:id)
- Enable Managed and verify the charts (/managed/summary)
- Go to NodeBalancers details and verify the charts (/nodebalancers/:id)

## As an Author I have considered 🤔

*Check all that apply*
